zcash_client_sqlite/src/wallet/db.rs

```diff
@@ -1095,7 +1105,10 @@ WITH unioned AS (
            NULL                         AS diversifier_index_be,
            sent_notes.value             AS value,
            0                            AS is_change,
-           sent_notes.memo              AS memo
+           sent_notes.memo              AS memo,
+           1                            AS is_trusted,
+           NULL                         AS t_input_mined_height,
+           1                            AS t_input_trust_status
```

Comment:
Bug: `t_input_trust_status` is only fixed to `true` for wallet-created transactions that solely spend wallet-created outputs. If the transaction spends inputs that were received externally, then `is_trusted` is fixed to `true` but `t_input_trust_status` would be false (unless you have a more subtle semantic meaning for `t_input_trust_status` than what I am interpreting it as, which I don't know because it is undocumented).

`t_input_mined_height` should not be fixed to `NULL` unless it is documented that this is the case for wallet-created outputs. But if the `t_input_trust_status` bug is fixed with joins, then we can trivially look up the input mined height same as for the other side of the union.

Reply:
All that being said, it's been long enough now since I worked on this PR that I no longer remember why it was that it alters `v_tx_outputs` in the first place.